src/platform/graphics/texture.js

```diff
@@ -1000,8 +1003,34 @@ class Texture {
         await Promise.all(promises);
     }
 
-    read(x, y, width, height, mipLevel = 0, face = 0, data = null, immediate = false) {
-        return this.impl.read?.(x, y, width, height, mipLevel, face, data, immediate);
+    /**
+     * Download the textures data from the graphics memory to the local memory.
+     *
+     * Note a public API yet, as not all options are implemented on all platforms.
+     *
+     * @param {number} x - The left edge of the rectangle.
+     * @param {number} y - The top edge of the rectangle.
+     * @param {number} width - The width of the rectangle.
+     * @param {number} height - The height of the rectangle.
+     * @param {object} [options] - Object for passing optional arguments.
+     * @param {number} [options.renderTarget] - The render target using the texture as a color
+     * buffer. Provide as an optimization to avoid creating a new render target. Important especially
+     * when this function is called with high frequency (per frame). Note that this is only utilized
+     * on the WebGL platform, and ignored on WebGPU.
+     * @param {number} [options.mipLevel] - The mip level to download. Defaults to 0.
+     * @param {number} [options.face] - The face to download. Defaults to 0.
+     * @param {Uint8Array|Uint16Array|Uint32Array|Float32Array} [options.data] - The data buffer to
+     * write the pixel data to. If not provided, a new buffer will be created. The type of the buffer
+     * must match the texture's format.
+     * @param {boolean} [options.immediate] - If true, the read operation will be executed as soon as
+     * possible. This has a performance impact, so it should be used only when necessary. Defaults
+     * to false.
+     * @returns {Promise<Uint8Array|Uint16Array|Uint32Array|Float32Array>} A promise that resolves
+     * with the pixel data of the texture.
+     * @ignore
+     */
+    read(x, y, width, height, options = {}) {
```

Comment:
Is `read` meant to supersede `downloadAsync`? Seems there is some duplication here. Couldn't there just be a single function that does it all?

Also, I prefer the name `downloadAsync` (to match `upload`) more than `read`.

Reply:
yes, downloadAsync is "deprecated" and will be removed. I left it there as there likely is some user of this.

`read` matches `StorageBuffer` `read` and `write`, `download` seems more internet term which seems confusing to use on the Texture.

Reply:
I removed `downloadAsync` - only the model viewer seems to be using it, and we can easily swap it to use `read` when we update it to the new engine

Reply:
So you'll rename `upload` to `write` as well? You think the functions `read`/`write` describes what is happening?

Reply:
`write` function would take user data, set it on the object and upload internally.
`upload` just uploads already provided data
`read` rives user a copy of data for reading

I think `read` is a pretty descriptive and simple name for what the user wants to do - read the content of the texture. Note that for now this is a non-public API, and we can revise this in a follow up PRs.